components/scream/scripts/test_all_scream.py

```diff
@@ -254,7 +255,6 @@ def __init__(self, cxx_compiler, f90_compiler, c_compiler, submit=False, paralle
                 print("test {} can use {} jobs to compile, and {} jobs for testing".format(test,self._compile_res_count[test],self._testing_res_count[test]))
 
         if self._keep_tree:
-            expect(not is_repo_clean(silent=True), "Makes no sense to use --keep-tree when repo is clean")
```

Comment:
I think it's a bit overly cautious for the program to fail in this case, but I could understand either way. The important thing is that I need scripts-tests to be robust to the state of the repo, so I have to add `-k` to all test-all-scream calls and have it work in both the dirty and clean repo case, so that's a strong motivation to have it be more tolerant.